gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/DagActionStoreChangeMonitor.java

```diff
@@ -136,6 +142,23 @@ protected void initializeMonitor() {
     }
   }
 
+  /*
+   Override this method to do the same sequence as the parent class, except create metrics. Instead, we create metrics
+   earlier upon class initialization because they are used immediately as dag actions are loaded and processed from
+   the DagActionStore.
+  */
+  @Override
+  protected void startUp() {
```

Comment:
Can we just move buildMetricsContextAndMetrics() to the constructor of HighLevelConsumer and avoid overriding startUp?

Reply:
A lot of existing tests need to be refactored if I do this and may end up having some unintended consequences bc HighLevelConsumer is used in many places. I'd rather not invest the time updating tests and debugging changes.